packages/jellyfish-api-core/__tests__/category/account/futureswap.test.ts

```diff
@@ -0,0 +1,922 @@
+    await testing.generate(1)
+    const attributes = await testing.rpc.masternode.getGov(attributeKey)
+    expect(attributes.ATTRIBUTES['v0/params/dfip2203/active']).toStrictEqual('false')
+
+    const fswap: FutureSwap = {
+      address: tslaAddress,
+      amount: `${swapAmount}@TSLA`
+    }
+
+    const promise = testing.rpc.account.futureSwap(fswap)
+
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test DFIP2203Tx execution failed:\nDFIP2203 not currently active\', code: -32600, method: futureswap')
+  })
+
+  it('should refund the futureswap if DFIP2203 is disabled before execution', async () => {
+    const swapAmount = 1
+    const tslaAddress = await testing.generateAddress()
+    await testing.rpc.account.accountToAccount(collateralAddress, { [tslaAddress]: `${swapAmount}@TSLA` })
+    await testing.generate(1)
+
+    // create the futureswap
+    const fswap: FutureSwap = {
+      address: tslaAddress,
+      amount: `${swapAmount}@TSLA`
+    }
+    await testing.rpc.account.futureSwap(fswap)
+    await testing.generate(1)
+
+    const nextSettleBlock = await testing.container.call('getfutureswapblock', [])
+
+    // check the futureswap is in effect
+    const pendingFutures = await testing.container.call('listpendingfutureswaps')
+    expect(pendingFutures.length).toStrictEqual(1)
+
+    // get minted DUSD
+    const dusdMintedBefore = (await testing.rpc.token.getToken(idDUSD))[idDUSD].minted
+
+    // deactivate DFIP2203
+    await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/active': 'false' } })
+    await testing.generate(1)
+    const attributes = await testing.rpc.masternode.getGov(attributeKey)
+    expect(attributes.ATTRIBUTES['v0/params/dfip2203/active']).toStrictEqual('false')
+
+    // move to nextSettleBlock
+    await testing.generate(nextSettleBlock - await testing.rpc.blockchain.getBlockCount())
+
+    // check futureswap is not executed.
+    {
+      const dusdMintedAfter = (await testing.rpc.token.getToken(idDUSD))[idDUSD].minted
+      expect(dusdMintedAfter).toStrictEqual(dusdMintedBefore)
+
+      const pendingFutures = await testing.container.call('listpendingfutureswaps')
+      expect(pendingFutures.length).toStrictEqual(0)
+
+      // check live/economy/dfip2203_*
+      const attributes = await testing.rpc.masternode.getGov(attributeKey)
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_current']).toStrictEqual([])
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_burned']).toBeUndefined()
+      expect(attributes.ATTRIBUTES['v0/live/economy/dfip2203_minted']).toBeUndefined()
+
+      {
+        // check contractAddress
+        const balance = await testing.rpc.account.getAccount(contractAddress)
+        expect(balance).toStrictEqual([])
+      }
+
+      {
+        // check tslaAddress
+        const balance = await testing.rpc.account.getAccount(tslaAddress)
```

Comment:
As mentioned refund, I think you should add another getAccount RPC somewhere after futureswap call. So we can see the difference of balance